packages/main/src/plugin/index.ts

```diff
@@ -1128,57 +1141,138 @@ export class PluginSystem {
         cancellableTokenId?: number,
         buildargs?: { [key: string]: string },
       ): Promise<unknown> => {
+        // create task
+        const task = taskManager.createTask({
+          title: `Build ${imageName}`,
+          action: {
+            name: 'Go to task >',
+            execute: () => {
+              navigationManager.navigateToImageBuild().catch((err: unknown) => {
+                console.error(`Something went wrong while trying to navigate to image build: ${String(err)}`);
+              });
+            },
+          },
+        });
+
         const abortController = this.createAbortControllerOnCancellationToken(
           cancellationTokenRegistry,
           cancellableTokenId,
         );
-        return containerProviderRegistry.buildImage(
-          containerBuildContextDirectory,
-          (eventName: string, data: string) => {
-            this.getWebContentsSender().send(
-              'container-provider-registry:buildImage-onData',
-              onDataCallbacksBuildImageId,
-              eventName,
-              data,
-            );
-          },
-          {
-            containerFile: relativeContainerfilePath,
-            tag: imageName,
-            platform,
-            provider: selectedProvider,
-            abortController,
-            buildargs,
-          },
-        );
+        return containerProviderRegistry
+          .buildImage(
+            containerBuildContextDirectory,
+            (eventName: string, data: string) => {
+              this.getWebContentsSender().send(
+                'container-provider-registry:buildImage-onData',
+                onDataCallbacksBuildImageId,
+                eventName,
+                data,
+              );
+            },
+            {
+              containerFile: relativeContainerfilePath,
+              tag: imageName,
+              platform,
+              provider: selectedProvider,
+              abortController,
+              buildargs,
+            },
+          )
+          .then(result => {
+            task.state = 'success';
+            return result;
+          })
+          .catch((err: unknown) => {
+            task.error = `Something went wrong while trying to build image: ${String(err)}`;
```

Comment:
Should we also set `task.state = 'error'` ? If true, this comments is related to all similar modifications in this file

Reply:
No need, if you set the property `error` it automatically set the sate to `error`

https://github.com/containers/podman-desktop/blob/3863310317f38f420a1981a8572ab1ada884e70f/packages/main/src/plugin/tasks/task-impl.ts#L69-L75